microsoft/hyper-v/default.nix

```diff
@@ -0,0 +1,104 @@
+{ config, pkgs, options, lib, ... }:
+
+with lib; {
+
+  options = {
+    hyperv.enhancedSession = mkOption
+      {
+        type = types.bool;
+        default = true;
+        description = ''
+          Enables enhanced hyperv session over RDP
+          Enable Guest Services on Hyper-V Management settings and
+          don't forget to enable access over VSock on a Admin powershell with:
+
+          % Set-Vm -VmName <VM> -EnhancedSessionTransportType HvSocket
+        '';
+      };
+
+    hyperv.video = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enables Display Manager support on HyperV
+      '';
+    };
+
+    hyperv.defaultWindowManager = mkOption
+      {
+        type = types.separatedString " ";
+        default = "${config.services.xserver.displayManager.sessionData.wrapper}";
+        description = ''
+          If ~/startwm.sh exists as an executable script, it will take priority over this option.
+        '';
+      };
+  };
+
+  config = mkMerge ([
+    {
+      boot.kernelModules = [ "hv_sock" ];
+      virtualisation.hypervGuest.enable = true;
+    }
+
+    (mkIf config.hyperv.video {
+      services.xserver = {
+        modules = [ pkgs.xorg.xf86videofbdev ];
+        videoDrivers = [ "hyperv_fb" ];
+      };
+    })
+
+    (mkIf
+      config.hyperv.enhancedSession
+      {
+        services.xrdp = {
+          enable = true;
+          defaultWindowManager = config.hyperv.defaultWindowManager;
+          package = pkgs.xrdp.overrideAttrs (old: rec {
+            configureFlags = old.configureFlags ++ [ " --enable-vsock" ];
+
+            postInstall = old.postInstall + ''
+              ${pkgs.gnused}/bin/sed -i -e 's!use_vsock=false!use_vsock=true!g'                               $out/etc/xrdp/xrdp.ini
```

Comment:
Do you see any way to move this configuration outside of the module? Than we don't need a rebuild to change options.

Reply:
I'm new to nix, so I'm not sure. On other systems, we edit a file on `/etc/xrdp` to make it work, but I'm not sure how to make it on nix

Reply:
Does xrdp maybe supports reading this configuration by supplying command line flags? Otherwise it might be possible to add configure flags in order to change sysconfdir (that's what it is called in autotools-based projects) to point to `/etc/xrdp` instead.